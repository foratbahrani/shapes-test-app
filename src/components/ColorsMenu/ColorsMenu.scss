.colorsMenu {
	display: flex;
	gap: 12px;
}
