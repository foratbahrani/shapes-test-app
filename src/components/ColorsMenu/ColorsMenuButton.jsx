import React from 'react';
import Shape from '../Shape/Shape';

const ColorsMenuButton = ({ color }) => {
	return <Shape key={color} color={color} shape='square' size='sm' />;
};

export default ColorsMenuButton;
