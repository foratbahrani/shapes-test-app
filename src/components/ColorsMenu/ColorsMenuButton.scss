.colorsMenuButton {
	appearance: none;
	border: none;
	background-color: transparent;
	cursor: pointer;
	padding: 0;
	margin: 0 8px;
	height: 30px;
}

.colorsMenuButton:focus {
	outline: 0;
}
