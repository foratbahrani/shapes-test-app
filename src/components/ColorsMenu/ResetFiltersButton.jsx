import React from 'react';
import { connect } from 'react-redux';
import { resetFilters } from '../../store/shapes/shapes.actions';
import Shape from '../Shape/Shape';
import './ColorsMenuButton.scss';

export const ResetFiltersButton = ({ color, resetFilters }) => {
	return (
		<button className={`colorsMenuButton`} id='button' onClick={() => resetFilters()}>
			<Shape key={color} shape='square' size='sm' />
		</button>
	);
};

const mapDispatchToProps = {
	resetFilters: resetFilters,
};

export default connect(null, mapDispatchToProps)(ResetFiltersButton);
