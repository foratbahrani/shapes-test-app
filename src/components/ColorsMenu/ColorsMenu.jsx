import React from 'react';
import './ColorsMenu.scss';
import ColorsMenuButton from './ColorsMenuButton';
import ResetFiltersButton from './ResetFiltersButton';

const ColorsMenu = () => {
	const colorsList = ['purple', 'red', 'green', 'blue', 'yellow'];
	const colorsComponents = colorsList.map(color => (
		<ColorsMenuButton key={color} color={color} />
	));
	return (
		<div className='colorsMenu'>
			{colorsComponents}
			<ResetFiltersButton />
		</div>
	);
};

export default ColorsMenu;
