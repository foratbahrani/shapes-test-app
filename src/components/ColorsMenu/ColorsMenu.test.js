import { shallow } from 'enzyme';
import React from 'react';
import ColorsMenu from './ColorsMenu';

it('matches snapshot', () => {
	expect(shallow(<ColorsMenu />)).toMatchSnapshot();
});
