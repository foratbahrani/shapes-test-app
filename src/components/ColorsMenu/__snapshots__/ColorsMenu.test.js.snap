// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="colorsMenu"
>
  <Connect(ColorsMenuButton)
    color="purple"
    key="purple"
  />
  <Connect(ColorsMenuButton)
    color="red"
    key="red"
  />
  <Connect(ColorsMenuButton)
    color="green"
    key="green"
  />
  <Connect(ColorsMenuButton)
    color="blue"
    key="blue"
  />
  <Connect(ColorsMenuButton)
    color="yellow"
    key="yellow"
  />
  <Connect(ResetFiltersButton) />
</div>
`;
